### PR TITLE
Add OCM pull model integration support by importing clusters without …

### PR DESCRIPTION
**Pull request by mikeshng**
…the managed cluster secret config data. Updated project to be more compatible with upstream ArgoCD and not just Openshift GitOps operator.

Signed-off-by: Mike Ng <ming@redhat.com>

/hold

### CODE REVIEW

File: pkg/controller/gitopscluster/gitopscluster_controller.go [hunk 1]
```diff
@@ -714,47 +720,89 @@ func (r *ReconcileGitOpsCluster) AddManagedClustersToArgo(
...
-		StringData: map[string]string{
-			"config": string(managedClusterSecret.Data["config"]),
-			"name":   string(managedClusterSecret.Data["name"]),
-			"server": string(managedClusterSecret.Data["server"]),
-		},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      managedCluster + "-cluster-secret",
+				Namespace: argoNamespace,
+				Labels: map[string]string{
+					"argocd.argoproj.io/secret-type":                 "cluster",
```
Comment (xiangjingli) [inline pkg/controller/gitopscluster/gitopscluster_controller.go:738]: When enablePullModel is on, we don't need to set up the label `argocd.argoproj.io/secret-type`. Or the dummy cluster  will be shown in the Argocd application UI  destination cluster dropdown list.

Comment (mikeshng) [reply pkg/controller/gitopscluster/gitopscluster_controller.go:738]: ApplicationSet controller will not be able to generate Application for these pull model managed clusters if there are no secrets with this label.

I think this is a known issue and we will have to wrap up all our pull model work and present to the ArgoCD community and ask how to tackle some of these issues.

Comment (xiangjingli) [reply pkg/controller/gitopscluster/gitopscluster_controller.go:738]: agree

**Comment by xiangjingli** [comment]
@mikeshng   one case we may need to consider to support - some clusters could be imported to support push model and pull model as well e.g.

1. cluster1 is imported to argocd NS first to support push model. Then there is a real cluster secret there
2. Later cluster1 is selected to support pull model. Then another dummy cluster secret is created.

As a result, we will have two cluster secrets for cluster1. I am not sure which cluster secret will be picked up by the ArgoCD applicationSet controller when propagating the ArgoCD applications.  

So ideally ArgoCD applicationSet controller should apply real cluster secrets to propagate ArgoCD applications in the push model, apply dummy cluster secrets in the pull model.

There may be some issues here. We may run some tests to verify this.

**Pull request merged**